### DISCUSSION TITLE
[V5] Scheduler issue with deployment

### DISCUSSION THREAD

**Discussion by pk4all** [adonisjs/core#3044] [Help] [answered]
Hello Team,
I am using "node-schedule" for my scheduled task. Every thing is working great.
But I have a issues with my project  deployment structure. We are using multi server structure with load balancer, so my project multiple instance running on multiple server, my issues is start now, My scheduler code running on all instance, it's create issues for me, some time same db row updated multiple time, it's possible with current situation.

I want to run my scheduler  task only one server, how this possible?
Sorry for my bad English. 

@thetutlage @RomainLanz 
Thanks.

**Comment by McSneaky** [comment, marked as answer]
That's the hard part of distributed systems :yum: 

It's long topic. Literally, there are multiple books and talks and research papers covering it.

Since you mentioned that you run into issues where multiple schedulers change same DB row, look into DB locks and do some pre-checking before starting new job (check if some worker has already pick up job before starting to work on that job)

Pseudocode:

```ts
// Get first non-running job and update it's `running` status
let job = await Job.where({ running: false }).first().update({ running: true })

await Database.lock(things_that_need_to_be_frozen_so_other_jobs_cant_edit_them)

/// Work, work, work

// Update database rows that need to be updated
await UpdateRequiredDatabaseRows()

// Remove lock, so rows become editable again
await Database.unlock(things_that_need_to_be_frozen_so_other_jobs_cant_edit_them)

// Delete job
await job.delete()
```

**Reply by pk4all** [reply]
Thanks a lot it's more helpful for me.
